docs/runtime/go-metrics.md

```diff
@@ -0,0 +1,218 @@
+
+**[1]:** Computed from `/memory/classes/heap/released:bytes`.
+<!-- endsemconv -->
+
+<!-- semconv metric.go.memory.released(full) -->
+<!-- endsemconv -->
+
+### Metric: `go.memory.limit`
+
+This metric is [recommended][MetricRecommended].
+
+<!-- semconv metric.go.memory.limit(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
+| -------- | --------------- | ----------- | -------------- | --------- |
+| `go.memory.limit` | UpDownCounter | `By` | Go runtime memory limit configured by the user, if a limit exists. [1] | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+
+**[1]:** Computed from `/gc/gomemlimit:bytes`. This metric is excluded if the limit obtained from the Go runtime is math.MaxInt64.
+<!-- endsemconv -->
+
+<!-- semconv metric.go.memory.limit(full) -->
+<!-- endsemconv -->
+
+### Metric: `go.memory.allocated`
+
+This metric is [recommended][MetricRecommended].
+
+<!-- semconv metric.go.memory.allocated(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
+| -------- | --------------- | ----------- | -------------- | --------- |
+| `go.memory.allocated` | Counter | `By` | Memory allocated to the heap by the application. [1] | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+
+**[1]:** Computed from `/gc/heap/allocs:bytes`.
+<!-- endsemconv -->
+
+<!-- semconv metric.go.memory.allocated(full) -->
+<!-- endsemconv -->
+
+### Metric: `go.memory.allocations`
+
+This metric is [recommended][MetricRecommended].
+
+<!-- semconv metric.go.memory.allocations(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
+| -------- | --------------- | ----------- | -------------- | --------- |
+| `go.memory.allocations` | Counter | `{allocation}` | Count of allocations to the heap by the application. [1] | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+
+**[1]:** Computed from `/gc/heap/allocs:objects`.
+<!-- endsemconv -->
+
+<!-- semconv metric.go.memory.allocations(full) -->
+<!-- endsemconv -->
+
+## Go Garbage Collection
+
+**Description:** Go metrics captured under the namespace `go.memory.gc.*`
+
+### Metric: `go.memory.gc.goal`
+
+This metric is [recommended][MetricRecommended].
+
+<!-- semconv metric.go.memory.gc.goal(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
+| -------- | --------------- | ----------- | -------------- | --------- |
+| `go.memory.gc.goal` | UpDownCounter | `By` | Heap size target for the end of the GC cycle. [1] | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+
+**[1]:** Computed from `/gc/heap/goal:bytes`.
+<!-- endsemconv -->
+
+<!-- semconv metric.go.memory.gc.goal(full) -->
+<!-- endsemconv -->
+
+### Metric: `go.memory.gc.user_goal`
```

Comment:
Could this just be called `gogc` or `gcpercent`? The concept is fairly specific, so calling it "goal" feels misleading to me.

Reply:
Yeah, I agree we need a better name.  I was almost thinking it might be better to put metrics which track user-provided configuration in the OTel resource if they are constant for the lifetime of the program, or group them together in a single metric (e.g. `go.config` with labels `gogc = 100, gomemlimit=12345`).  Can gogc or gomemlimit be modified during the execution of the program?

Reply:
Yeah, both can be modified during execution, and possibly with some frequency (the API is fairly responsive by design). That's another reason that exporting it is useful (though changing it frequently is less common). The relevant APIs are `runtime/debug.SetGCPercent` and `runtime/debug.SetMemoryLimit` for `GOGC` and `GOMEMLIMIT` respectively.

Reply:
updated to `go.config.gogc`.  I don't love that it isn't consistent with the memory limit (which is still go.memory.limit), but I think it is at least clearer what the metric is.